### DISCUSSION TITLE
How to make Messagebox.show_warning popup window centered on the parent window?

### DISCUSSION THREAD

**Discussion by zhenzi0322** [israel-dryer/ttkbootstrap#345] [Q&A]
How to make Messagebox.show_warning popup window centered on the parent window?

```python
import ttkbootstrap as ttk
from tkinter import messagebox
from ttkbootstrap.constants import *
from ttkbootstrap.dialogs.dialogs import Messagebox

root = ttk.Window()
root.wm_minsize(1280, 700)


def open_ttkbootstrap_warning():
    Messagebox.show_warning("Hello")


def open_tkinter_warning():
    messagebox.showwarning(message="Hello")


b1 = ttk.Button(root, text="ttkbootstrap show_warning", bootstyle=SUCCESS, command=open_ttkbootstrap_warning)
b1.pack(side=LEFT, padx=5, pady=10)

b2 = ttk.Button(root, text="tkinter show_warning", bootstyle=(INFO, OUTLINE), command=open_tkinter_warning)
b2.pack(side=LEFT, padx=5, pady=10)

root.mainloop()
```

**Comment by jiangwu007** [comment]
def open_ttkbootstrap_warning():
    Messagebox.show_warning("Hello", position=(root.winfo_x() + root.winfo_width() // 2 - 100,
                                               root.winfo_y() + root.winfo_height() // 2 - 100))
你可以这样写,获取主窗口的x,y坐标与长宽.

**Reply by zhenzi0322** [reply]
> def open_ttkbootstrap_warning(): Messagebox.show_warning("Hello", position=(root.winfo_x() + root.winfo_width() // 2 - 100, root.winfo_y() + root.winfo_height() // 2 - 100)) 你可以这样写,获取主窗口的x,y坐标与长宽.

谢谢。这种可以实现。还有一个问题就是弹窗的图标为什么没有变的呢？或者是怎么设置弹窗的图标。
